### PR TITLE
Add wait for gathering threshold signatures instead of all signatures

### PR DESCRIPTION
**Pull request by abramsymons**

### CODE REVIEW

File: common/bls.py [hunk 1]
```diff
@@ -65,8 +65,38 @@ async def gather_and_aggregate_signatures(
             )
         )
         for node_id in node_ids
-    ]
-    signatures: list[dict[str, Any] | None] = await asyncio.gather(*tasks)
+    }
+
+    aggregate_timeout = 10 # in seconds
```
Comment (abramsymons) [inline common/bls.py:70]: This should be read from config

Comment (mchitgarha78) [reply common/bls.py]: Resolved in this [commit](https://github.com/zellular-xyz/zsequencer/commit/c4b6c2088997246e75916014a9b12b18b28c0263)

File: config.py [hunk 2]
```diff
@@ -91,6 +92,9 @@ def load_environment_variables(self):
         self.FINALIZATION_TIME_BORDER: int = int(
             os.getenv("ZSEQUENCER_FINALIZATION_TIME_BORDER", "120")
         )
+        self.AGGREGATION_TIMEOUT: int = int(
+            os.getenv("ZSEQUENCER_SIGNATURES_AGGREGATION_TIMEOUT", "120")
```
Comment (abramsymons) [inline config.py:96]: Let's reduce this default value to 5 seconds

File: examples/runner.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 ZSEQUENCER_SEND_TXS_INTERVAL: float = 0.01
 ZSEQUENCER_SYNC_INTERVAL: float = 0.01
 ZSEQUENCER_FINALIZATION_TIME_BORDER: int = 120
+ZSEQUENCER_SIGNATURES_AGGREGATION_TIMEOUT = 10
```
Comment (abramsymons) [inline examples/runner.py:26]: Let's reduce this default value to 5 seconds

Comment (mchitgarha78) [reply examples/runner.py]: Fixed.

File: config.py (hunk 2 above)
Comment (mchitgarha78) [reply config.py]: Fixed.

**Review by abramsymons — COMMENTED** [review]
Thank you

**Pull request merged**